getFunctionalInterfaceMethod() fails to recognize method override

On the latest master (f31a2f7). Here's a patch to produce a failing test:

```diff
diff --git a/nullaway/src/test/java/com/uber/nullaway/NullAwayTest.java b/nullaway/src/test/java/com/uber/nullaway/NullAwayTest.java
index e318ff0..1863775 100644
--- a/nullaway/src/test/java/com/uber/nullaway/NullAwayTest.java
+++ b/nullaway/src/test/java/com/uber/nullaway/NullAwayTest.java
@@ -139,4 +139,9 @@ public class NullAwayTest {
     public void functionalMethodSuperInterface() {
         compilationHelper.addSourceFile("NullAwaySuperFunctionalInterface.java").doTest();
     }
+
+    @Test
+    public void functionalMethodOverrideSuperInterface() {
+        compilationHelper.addSourceFile("NullAwayOverrideFunctionalInterfaces.java").doTest();
+    }
 }
diff --git a/nullaway/src/test/resources/com/uber/nullaway/testdata/NullAwayOverrideFunctionalInterfaces.java b/nullaway/src/test/resources/com/uber/nullaway/testdata/NullAwayOverrideFunctionalInterfaces.java
index e69de29..9383bef 100644
--- a/nullaway/src/test/resources/com/uber/nullaway/testdata/NullAwayOverrideFunctionalInterfaces.java
+++ b/nullaway/src/test/resources/com/uber/nullaway/testdata/NullAwayOverrideFunctionalInterfaces.java
@@ -0,0 +1,27 @@
+package com.uber.nullaway.testdata;
+
+import java.util.function.ToIntFunction;
+
+public class NullAwayOverrideFunctionalInterfaces {
+
+    public void test() {
+        call(str -> 42);
+    }
+
+    private int call(ObjToInt<String> f) {
+        return f.call("The answer to life the universe and everything");
+    }
+
+    @FunctionalInterface
+    private static interface ObjToInt<T> extends ObjToIntE<T, RuntimeException>, ToIntFunction<T> {
+        @Override
+        default int applyAsInt(T t) {
+            return call(t);
+        }
+    } 
+
+    @FunctionalInterface
+    private static interface ObjToIntE<T, E extends Exception> {
+        int call(T t) throws E;
+    }
+}
```

Here's the relevant portion of the error stacktrace:

```
java.lang.AssertionError: An unhandled exception was thrown by the Error Prone static analysis plugin.
     Please report this at https://github.com/google/error-prone/issues/new and include the following:
  
     error-prone version: 2.1.1
     Stack Trace:
     java.lang.RuntimeException: already found an answer! call(T) applyAsInt(T) applyAsInt(T)
  	at com.uber.nullaway.NullabilityUtil.getFunctionalInterfaceMethod(NullabilityUtil.java:87)
  	at com.uber.nullaway.NullAway.matchLambdaExpression(NullAway.java:422)
```

--- a/nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java
+++ b/nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java
@@ -64,36 +64,7 @@ public class NullabilityUtil {
      */
     public static Symbol.MethodSymbol getFunctionalInterfaceMethod(LambdaExpressionTree tree, Types types) {
         Type funcInterfaceType = ((JCTree.JCLambda) tree).type;
-
-        Symbol.MethodSymbol result = null;
-        for (Type t : types.closure(funcInterfaceType)) {
-            // we want the method symbol for the single function inside the interface...hrm
-            List<Symbol> enclosedElements = t.tsym.getEnclosedElements();
-            for (Symbol s : enclosedElements) {
-                Symbol.MethodSymbol elem = (Symbol.MethodSymbol) s;
-                // The only constructor we should be seeing here is Object(), since these are all interfaces.
-                // Nonetheless, we need to filter that one with isConstructor()
-                if (elem.isDefault() || elem.isStatic() || elem.isConstructor() ) {
-                    continue;
-                }
-                String name = elem.getSimpleName().toString();
-                // any methods overridding java.lang.Object methods don't count;
-                // see https://docs.oracle.com/javase/8/docs/api/java/lang/FunctionalInterface.html
-                // we should really be checking method signatures here; hack for now
-                if (OBJECT_METHOD_NAMES.contains(name)) {
-                    continue;
-                }
-                if (result != null) {
-                    throw new RuntimeException(
-                            "already found an answer! " + result + " " + elem + " " + enclosedElements);
-                }
-                result = elem;
-            }
-        }
-        if (result == null) {
-            throw new RuntimeException("could not find functional interface method for " + funcInterfaceType);
-        }
-        return result;
+        return (Symbol.MethodSymbol) types.findDescriptorSymbol(funcInterfaceType.tsym);
     }
 
     /**

--- a/nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java
+++ b/nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java
@@ -22,7 +22,6 @@
 
 package com.uber.nullaway;
 
-import com.google.common.collect.ImmutableSet;
 import com.google.errorprone.util.ASTHelpers;
 import com.sun.source.tree.LambdaExpressionTree;
 import com.sun.source.tree.MethodTree;
@@ -34,26 +33,12 @@ import com.sun.tools.javac.code.Types;
 import com.sun.tools.javac.tree.JCTree;
 import com.sun.tools.javac.util.JCDiagnostic;
 
-import java.util.List;
-
 import javax.annotation.Nullable;
 
 /**
  * Helpful utility methods for nullability analysis.
  */
 public class NullabilityUtil {
-
-    static final ImmutableSet<String> OBJECT_METHOD_NAMES = ImmutableSet.of(
-            "equals",
-            "hashCode",
-            "toString",
-            "finalize",
-            "clone",
-            "notify",
-            "notifyAll",
-            "wait",
-            "getClass"
-    );
 
     private NullabilityUtil() { }
 

--- a/nullaway/src/test/java/com/uber/nullaway/NullAwayTest.java
+++ b/nullaway/src/test/java/com/uber/nullaway/NullAwayTest.java
@@ -139,4 +139,9 @@ public class NullAwayTest {
     public void functionalMethodSuperInterface() {
         compilationHelper.addSourceFile("NullAwaySuperFunctionalInterface.java").doTest();
     }
+
+    @Test
+    public void functionalMethodOverrideSuperInterface() {
+        compilationHelper.addSourceFile("NullAwayOverrideFunctionalInterfaces.java").doTest();
+    }
 }

--- a/nullaway/src/test/resources/com/uber/nullaway/testdata/NullAwayOverrideFunctionalInterfaces.java
+++ b/nullaway/src/test/resources/com/uber/nullaway/testdata/NullAwayOverrideFunctionalInterfaces.java
@@ -1,0 +1,27 @@
+package com.uber.nullaway.testdata;
+
+import java.util.function.ToIntFunction;
+
+public class NullAwayOverrideFunctionalInterfaces {
+
+    public void test() {
+        call(str -> 42);
+    }
+
+    private int call(ObjToInt<String> f) {
+        return f.call("The answer to life the universe and everything");
+    }
+
+    @FunctionalInterface
+    private static interface ObjToInt<T> extends ObjToIntE<T, RuntimeException>, ToIntFunction<T> {
+        @Override
+        default int applyAsInt(T t) {
+            return call(t);
+        }
+    }
+
+    @FunctionalInterface
+    private static interface ObjToIntE<T, E extends Exception> {
+        int call(T t) throws E;
+    }
+}

